src/crypto/public_key/dsa.js

```diff
@@ -85,8 +85,8 @@ export default {
       break;
     }
     return {
-      r: r.toArrayLike(Uint8Array),
-      s: s.toArrayLike(Uint8Array)
+      r: r.toArrayLike(Uint8Array, 'be', 32),
+      s: s.toArrayLike(Uint8Array, 'be', 32)
```

Comment:
Thanks for the PR! :+1: I think these should be `q.bitLength() / 32` instead of 32 (the signature size varies by key size).

Reply:
P.S. A bit unrelated but the two calls to `util.getLeftNBits(hashed, q.bitLength()))` could be replaced by `hashed.subarray(0, q.bitLength() / 8)`; q's bit length must be a multiple of 8. Then `getLeftNBits` can be removed. You don't have to do that in this PR of course, only if you feel like it.

Reply:
> Thanks for the PR! :+1: I think these should be `q.bitLength() / 32` instead of 32 (the signature size varies by key size).

Is there a way to produce failure with the currently implemented code `r.toArrayLike(Uint8Array, 'be', 32)` ? If yes, it would be wise to add a test specifically for that (with appropriate key size), as this kind of thing would be easy to miss in the future.

Reply:
Hmm. I think it would actually still work with this code, as 32 bytes is enough to fit every signature size. Nevertheless, I think we don't have any tests for that. I agree it would be nice to have one.

Reply:
Maybe not as crucial to have a test for this if it would not cause a hard failure (and is not a security concern). Thanks for the explanation :+1: